Lean: simplify usage of sed

ineol opened the pull request:
The sed interface in MacOS and in Linux is not compatible.

ineol committed:
Lean: simplify the usage of sed

Alasdair commented:
I think the workaround is something like `sed -i.bak` giving a file extension to the `-i` option. The lack of a space is required for GNU sed and macOS sed compatibility iirc.

ineol committed:
Lean: add coercions and use Lean native binops

We use the coercion system to cast bitvectors of any length.

We also use the native Lean +, -, etc because the elaborator
treats them in a special way when infering the types of the
result of the operation.

ineol commented:
Then there will be temp files left around right?

bacam commented:
Previously: https://github.com/riscv/sail-riscv/pull/177

ineol commented:
I see, so you prefer to use `sed` then `mv` as for the riscv repo?

bacam commented:
Maybe you could replace the `cp` with `sed` redirected to the target file?

ineol commented:
Yes, that way better thank you! I just did
